R/clean_garbage.R

```diff
@@ -0,0 +1,153 @@
+        return(NULL)
+      } else {
+        return(arrow_dir)
+      }
+    } else if (inherits(x, what = "list")) {
+
+      # Check if any list contains Seurat objects or CellGraphAssays
+      list_arrow_dirs <- lapply(x, function(y) {
+        if (inherits(y, what = c("Seurat", "CellGraphAssay"))) {
+          # Ignore Seurat objects without CellGraphAssays
+          arrow_dir <- try({
+            ArrowDir(y)
+          }, silent = TRUE)
+          if (inherits(arrow_dir, "try-error")) {
+            return(NULL)
+          } else {
+            return(arrow_dir)
+          }
+        } else {
+          return(NULL)
+        }
+      })
+      return(list_arrow_dirs %>% set_names(seq_along(list_arrow_dirs)))
+    } else {
+      return(NULL)
+    }
+  })
+
+  # Unlist and remove NULLs
+  global_variable_types <- unlist(global_variable_types)
+  global_variable_types <- Filter(Negate(is.null), global_variable_types)
+
+  # Convert to tibble
+  if (length(global_variable_types) > 0) {
+    global_variable_types <- tibble(variable = names(global_variable_types), edgelist_dir = global_variable_types)
+  }
+
+  # Get command log
+  command_log <- getOption("pixelatorR.edgelist_copies")
+  if (length(command_log) > 0) {
+    if (length(global_variable_types) > 0) {
+      command_log <- command_log %>%
+        left_join(global_variable_types, by = "edgelist_dir")
+    } else {
+      command_log <- NULL
+    }
+  }
+
+  # Get edgelist directories
+  edgelist_dirs <- list.files(getOption("pixelatorR.arrow_outdir"), full.names = TRUE) %>% normalizePath()
+  if (length(edgelist_dirs) > 0) {
+    edgelist_dirs <- tibble(edgelist_dir = edgelist_dirs, exists = TRUE)
+    if (length(global_variable_types) > 0) {
+      command_log <- command_log %>%
+        full_join(edgelist_dirs, by = "edgelist_dir")
+    } else {
+      command_log <- edgelist_dirs
+    }
+  } else {
+    cli_alert_info("Found no edgelist directories linked to missing variables")
+    return(invisible(NULL))
+  }
+
+  # Only keep existing directories
+  command_log <- command_log %>%
+    filter(dir.exists(edgelist_dir))
+
+  if (!"variable" %in% names(command_log)) {
+    command_log <- command_log %>% mutate(variable = NA_character_)
+  }
+
+  # Remove edgelists linked to missing variables
+  command_log_na <- command_log %>%
+    filter(is.na(variable))
+
+  # Run check on directories to make sure that they match the expected pattern
+  command_log_na <- command_log_na %>%
+    mutate(
+      pattern_match = stringr::str_detect(basename(edgelist_dir),
+                                          pattern = "[A-Za-z0-9]{5}-\\d{4}-\\d{2}-\\d{2}-\\d{6}")
+    ) %>%
+    filter(pattern_match)
+
+  # Exit if no directories are found
+  if (nrow(command_log_na) == 0) {
+    cli_alert_info("Found no edgelist directories linked to missing variables")
+    return(invisible(NULL))
+  }
+
+  if (getOption("pixelatorR.interactive", default = TRUE)) {
```

Comment:
Does this mean that files can't be cleaned up unless `pixelatorR` is run interactively?

Reply:
Maybe I should rename that option. If it's set to FALSE, it will not generate a prompt to ask if you want to delete the files. It will just delete them, basically the same as answering yes to the prompt.